src/internal/operators/zipWith.ts

```diff
@@ -1,48 +1,84 @@
+): OperatorFunction<T, R>;
 /** @deprecated Deprecated use {@link zipWith} */
 export function zip<T, T2>(v2: ObservableInput<T2>): OperatorFunction<T, [T, T2]>;
 /** @deprecated Deprecated use {@link zipWith} */
 export function zip<T, T2, T3>(v2: ObservableInput<T2>, v3: ObservableInput<T3>): OperatorFunction<T, [T, T2, T3]>;
 /** @deprecated Deprecated use {@link zipWith} */
-export function zip<T, T2, T3, T4>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>): OperatorFunction<T, [T, T2, T3, T4]>;
+export function zip<T, T2, T3, T4>(
+  v2: ObservableInput<T2>,
+  v3: ObservableInput<T3>,
+  v4: ObservableInput<T4>
+): OperatorFunction<T, [T, T2, T3, T4]>;
 /** @deprecated Deprecated use {@link zipWith} */
-export function zip<T, T2, T3, T4, T5>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>): OperatorFunction<T, [T, T2, T3, T4, T5]>;
+export function zip<T, T2, T3, T4, T5>(
+  v2: ObservableInput<T2>,
+  v3: ObservableInput<T3>,
+  v4: ObservableInput<T4>,
+  v5: ObservableInput<T5>
+): OperatorFunction<T, [T, T2, T3, T4, T5]>;
 /** @deprecated Deprecated use {@link zipWith} */
-export function zip<T, T2, T3, T4, T5, T6>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, v6: ObservableInput<T6>): OperatorFunction<T, [T, T2, T3, T4, T5, T6]> ;
+export function zip<T, T2, T3, T4, T5, T6>(
+  v2: ObservableInput<T2>,
+  v3: ObservableInput<T3>,
+  v4: ObservableInput<T4>,
+  v5: ObservableInput<T5>,
+  v6: ObservableInput<T6>
+): OperatorFunction<T, [T, T2, T3, T4, T5, T6]>;
 /** @deprecated Deprecated use {@link zipWith} */
 export function zip<T, R>(...observables: Array<ObservableInput<T> | ((...values: Array<T>) => R)>): OperatorFunction<T, R>;
 /** @deprecated Deprecated use {@link zipWith} */
 export function zip<T, R>(array: Array<ObservableInput<T>>): OperatorFunction<T, R>;
 /** @deprecated Deprecated use {@link zipWith} */
-export function zip<T, TOther, R>(array: Array<ObservableInput<TOther>>, project: (v1: T, ...values: Array<TOther>) => R): OperatorFunction<T, R>;
+export function zip<T, TOther, R>(
+  array: Array<ObservableInput<TOther>>,
+  project: (v1: T, ...values: Array<TOther>) => R
+): OperatorFunction<T, R>;
 /* tslint:enable:max-line-length */
 
 /**
  * @deprecated Deprecated. Use {@link zipWith}.
  */
-export function zip<T, R>(...sources: Array<ObservableInput<any> | ((...values: Array<any>) => R)>): OperatorFunction<T, R> {
-  return (source: Observable<T>) => lift(source, function (this: Subscriber<any>, source: Observable<T>) {
-    const args = [source, ...sources];
-    return zipStatic(...args).subscribe(this);
-  })
+export function zip<T, R>(...sources: Array<ObservableInput<any> | ((...values: Array<any>) => R)>): OperatorFunction<T, any> {
+  return operate((source, subscriber) => zipStatic(source, ...sources).subscribe(subscriber));
```

Comment:
I've updated `repeat` and `retry` to "modernize" them to our new way of doing things. They were an oversight in #5729. 

I've also updated the types in `operate` so you can only return `(() => void) | void`. This will help force us to do the right thing, and it caught several others that we didn't see. However, if we push this API public, I suspect there will be a lot of people that try to return `return () => subscription.unsubscribe()` there... which isn't ideal.